prow/pod-utils/gcs/upload.go

```diff
@@ -52,18 +53,23 @@ func LocalExport(exportDir string, uploadTargets map[string]UploadFunc) error {
 			filePath: path.Join(exportDir, dest),
 		}
 	}
-	return upload(dtw, uploadTargets)
+	return upload(dtw, uploadTargets, int64(len(uploadTargets))) // there is no use in limiting concurrency of local writes
 }
 
-func upload(dtw destToWriter, uploadTargets map[string]UploadFunc) error {
+func upload(dtw destToWriter, uploadTargets map[string]UploadFunc, maxConcurrency int64) error {
 	errCh := make(chan error, len(uploadTargets))
 	group := &sync.WaitGroup{}
 	group.Add(len(uploadTargets))
+	sem := semaphore.NewWeighted(maxConcurrency)
 	for dest, upload := range uploadTargets {
 		log := logrus.WithField("dest", dest)
+		if err := sem.Acquire(context.Background(), 1); err != nil {
+			log.WithError(err).Error("Could not begin upload.")
```

Comment:
Will having an unpaired release break/crash?

Reply:
How could it be unpaired?

Reply:
If acquire fails here, you still call the go func, which will call release?

Reply:
Ah, eek

Reply:
Should only fail if the context is cancelled, which it cannot be, but yes